src/errors.ts

```diff
@@ -29,6 +29,20 @@ export class ReadOnlyError extends AzFuncTypeError {
 
 export function ensureErrorType(err: unknown): Error & Partial<AzFuncError> {
     if (err instanceof Error) {
+        const writable = Object.getOwnPropertyDescriptor(err, 'message')?.writable;
+        if (!writable) {
+            // The motivation for this branch can be found in the below issue:
+            // https://github.com/Azure/azure-functions-nodejs-library/issues/205
+            let readableMessage = err.message;
+            Object.defineProperty(err, 'message', {
```

Comment:
We're seeing a TypeError that causes an unhandled promise rejection on this line of code and prevents the actual error message from being raised. Here's the TypeError:
```json
{
    "message": "Cannot redefine property: message",
    "name": "TypeError",
    "stack": "TypeError: Cannot redefine property: message\n    at Function.defineProperty (<anonymous>)\n    at t.ensureErrorType (C:\\Program Files (x86)\\SiteExtensions\\Functions\\4.34.1\\workers\\node\\dist\\src\\worker-bundle.js:2:31119)\n    at C:\\Program Files (x86)\\SiteExtensions\\Functions\\4.34.1\\workers\\node\\dist\\src\\worker-bundle.js:2:53526\n    at Generator.throw (<anonymous>)\n    at a (C:\\Program Files (x86)\\SiteExtensions\\Functions\\4.34.1\\workers\\node\\dist\\src\\worker-bundle.js:2:51940)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
}
```

Reply:
Maybe the error being thrown is sealed? I can reproduce the error we're seeing with this test code:
```javascript
const error = Object.seal(new Error('Sealed error'));
Object.defineProperty(error, 'message', { get: () => 'Redefined error message' });
```

Reply:
@ryanbecker to clarify are you worse off after we shipped this PR or just equally bad?

Reply:
I believe in this case we're worse off. We'll wrap all our function code in a try/catch to log the error that's causing this so we can better understand the root cause, but I expect the error object is sealed. Maybe the error doesn't need to be mutated [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/588e99fa3945b3cfa82d6f32aec91f96f1735cdc/src/eventHandlers/FunctionLoadHandler.ts#L46) or in this code? It might be difficult to ensure this works with any error that could be thrown.

Reply:
@ryanbecker I just tried out your test code. It's an interesting corner case. We do not enter this `if` block because the sealed object is in fact `writable`.  However, the problem is that it is not `configurable`. It seems we need to check that both these properties are set to `true`.

I found a way to do that with `structuredClone`. I will discuss with Eric first though. Thanks for raising this issue!

Reply:
@ryanbecker I opened a new PR for this issue: #750.